### PR TITLE
Log failures to register discoverable commands

### PR DESCRIPTION
**Pull request by Kraagen**

### CODE REVIEW

**Review by malibuworkcrew — APPROVED** [review]

**Comment by malibuworkcrew** [comment]
Looks good!

File: src/main/scala/com/webtrends/harness/component/zookeeper/discoverable/DiscoverableCommandHelper.scala [hunk 1]
```diff
@@ -24,59 +24,38 @@ trait DiscoverableCommandHelper extends CommandHelper with Discoverable {
...
-            }
-          case None => p failure CommandException("CommandManager", "CommandManager not found!")
-        }
-      case Failure(f) => p failure f
+    val idValue = id.getOrElse(UUID.randomUUID().toString)
+    val m = AddCommandWithProps(name, props)
+
+    val result = for {
+      _ <- initCommandManager
+      _ <- commandManager.map(_ ? m).getOrElse(Future.failed(new CommandException("CommandManager", "CommandManager not found!")))
```
Comment (AlexanderNull) [inline src/main/scala/com/webtrends/harness/component/zookeeper/discoverable/DiscoverableCommandHelper.scala]: previous implementation returned the result of this line, is it intended to switch up the return value here?

**Review by AlexanderNull — APPROVED** [review]
Looks good, would suggest a major bump however. The return type would qualify as a breaking interface change.

**Comment by AlexanderNull** [comment]
Also, thank you for getting rid of those horrid Option matches

**Pull request merged**